### DISCUSSION TITLE
Official support for character base trait identifiers! (Gender, age etc...)

### DISCUSSION THREAD

**Discussion by Kiwi1444** [ikemen-engine/Ikemen-GO#1987] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

Mugen has unofficial base trait identifiers by Messatsu, that some character uses. In simple terms, these are blank animations, by the presence of which, you can determine certain character traits. Here the animation number and the corresponding trait:

`8055000,8055999 - Standard Detection:
These two anims need to both be present in your character for detection purposes, to make sure he/she/it complies with these trait identifiers.`

`Gender:
8055001 - Male
8055002 - Female
8055003 - Neither`

`Age he/she/it appears to be:
8055010 - Baby
8055011 - Child
8055012 - Teenager
8055013 - Young Adult
8055014 - Middle Aged
8055015 - Old
8055016 - Ancient
8055017 - Timeless`

`Type:
8055020 - Human
8055021 - Robot
8055022 - Animal
8055023 - Monster
8055024 - Ghost
8055025 - Mech
8055026 - Mythological Creature
8055027 - Alien
8055028 - God
8055029 - Object`

`Blood Color:
8055030 - Red
8055031 - Green
8055032 - Blue
8055033 - Yellow
8055034 - Purple
8055035 - Orange
8055036 - Black
8055037 - White
8055038 - Grey
8055039 - Does not bleed`

`Armor Type / Body Composition (i.e. For hitsounds):
8055040 - None
8055041 - Metal
8055042 - Stone
8055043 - Wood
8055044 - Flame
8055045 - Ice
8055046 - Force`

`Blocking Armor Type (Think shields, for example):
8055047 - None
8055048 - Metal
8055049 - Stone
8055050 - Wood
8055051 - Flame
8055052 - Ice
8055053 - Force`

`Weapon:
8055054 - Unarmed
8055055 - Bladed
8055056 - Piercing
8055057 - Blunt
8055058 - Magic
8055059 - Projectile (guns, etc)`

`Attractiveness:
8055060 - God Awful
8055061 - Ugly
8055062 - Average
8055063 - Hot
8055064 - Cute/Dead Sexy`

`Coolness:
8055065 - Really Dorky
8055066 - A Little Dorky
8055067 - Normal
8055068 - Cool
8055069 - Very Cool`

`Morality:
8055070 - Evil
8055071 - Mean
8055072 - Neutral
8055073 - Good Natured
8055074 - Absolutely Moral (?!)`

`Intelligence:
8055080 - No Cognition
8055081 - Stupid
8055082 - Average
8055083 - Smart
8055084 - Genius
8055085 - Winane (^_-)`

`Artificial Inteligence:
8055090 - Is a Joke
8055091 - Easy to Beat
8055092 - Average
8055093 - Difficult to beat
8055094 - Boss
8055095 - Uber Cheap`

`Height:
8055100 - Tiny
8055101 - Midget
8055102 - Short
8055103 - Average Height
8055104 - Tall
8055105 - Towering`

`Physical Build:
8055110 - Ghastly Thin
8055111 - Slender
8055112 - Average
8055113 - Buff
8055114 - Massive
8055115 - Chubby
8055116 - Fat`

`Fighting Styles:
8055120 -  Aikido
8055121 - Boxing
8055122 - Capoeira
8055123 - Choy Li Fut
8055124 - Dim Mak
8055125 - Gojukai Karate
8055126 - Goju Ryu Karate
8055127 - Hapkido
8055128 - Hsing
8055129 - Iaido
8055130 - Jeet Kune Do
8055131 - Judo
8055132 - Jujutsu
8055133 - Ju Jitsu
8055134 - Kali
8055135 - Karate
8055136 - Kendo
8055137 - Kenpo Karate
8055138 - Kickboxing
8055139 - Krav Maga
8055140 - Kung Fu
8055141 - Kyokushinkai
8055142 - Muay Thai
8055143 - Ninjutsu
8055144 - Pentjak Silat
8055145 - Pa Kua
8055146 - Sambo
8055147 - Savate
8055148 - Shito Ryu Karate
8055149 - Shorin Ryu
8055150 - Shorinji Kempo
8055151 - Shotokan Karate
8055152 - Shukokai
8055153 - Sumo
8055154 - Taekwondo
8055155 - Tai Chi Chuan
8055156 - Tang Soo Do
8055157 - Uechi Ryu
8055158 - Wado Ryu Karate
8055159 - Wing Chun Kung Fu
8055160 - Yoseikan
8055161 - Ansatsukan
8055162 - Wrestling
8055163 - Drunken Boxing
8055164 - Ankoku Karate
8055165 - Koppo
8055166 - KOBUJUTSU
8055167 - SELF-TAUGHT FIGHTING
8055168 - WU SHU
8055169 - Unknown fighting style
8055170 - KITANTAN-RYUU
8055171 - DAINANRYU-AIKI JUUJUTSU
8055172 - Kenka Sappou
8055173 - Hizoku
8055174 - Vale Tudo
8055175 - Saikyo-ryuu
8055176 - Bushin-ryuu`

`Blood Types: Just for SF2 fun   Use both A and B for AB:
8055190 - A
8055191 - B
8055192 - O
8055193 - +
8055194 - -`

`Race: (pick multiple) uses US census list.  don't complain about how limited it is until you, I or Messatsu can find a comprehensive list (this proves rather difficult):
8055200 - White
8055201 - African
8055202 - Chinese
8055203 - American Indian
8055204 - Japanese
8055205 - Filipino
8055206 - Indian
8055207 - Korean
8055208 - Hawaiian
8055209 - Vietnamese
8055210 - Guamanian
8055211 - Samoan
8055212 - Latino`

 I would like official support for this system in Ikemen Go with the ability to specify this information in the def file and the corresponding triggers that understand it. E.g. in def file you prescribe: `Gender = male`

Then corresponding trigger `enemy,getidentifier(gender)` will return 1. Etc...

Optionally, these triggers can support detection by animation method. E. g. if character have 8055000 and 8055001 animations, then `enemy,getidentifier(gender)` will return 1 also.

### Search terms

character, trait

**Comment by lovelydumpling** [comment]
Can't these, and virtually anything else you might want identified, be handled through Maps? 
https://github.com/ikemen-engine/Ikemen-GO/wiki/Character-features#map

**Reply by Kiwi1444** [reply]
> Can't these, and virtually anything else you might want identified, be handled through Maps? https://github.com/ikemen-engine/Ikemen-GO/wiki/Character-features#map

This trigger looks like it would fit, but I'd like a consistent standard at least in terms of gender.

**Comment by potsmugen** [comment, marked as answer]
We don't need official support for those I think. People already have the tools to make them on their own.

**Reply by Kiwi1444** [reply]
Not everyone knows that such opportunities exist at all. On the other hand, a special column in the def or cns file will clearly show that such opportunities exist. It will be more user-friendly.

**Reply by lovelydumpling** [reply]
To do anything with that information would require a sufficiently advanced knowledge of character code anyway. Simply adding some gender flag doesn't do anything unless characters are designed specifically to interact with them in some way. There isn't a reason to make it a hardcoded option in engine.

**Reply by Kiwi1444** [reply]
Wouldn't it be cool that as many characters as possible would have special intros, victory phrases, or even special moves(!) that triggered by age or gender? All we need for this to work is for as many characters as possible to provide this information. My character BTW already have special intro and victory phrases dependent from opponent gender.
